uiTests/src/test/java/com/virtuslab/gitmachete/uitest/UiTestSuite.java

```diff
@@ -70,14 +68,18 @@ public void ensureCorrectGraphTableRowCounts() {
   }
 
   @Test
-  public void pullCurrentBranchAndEnsureCleanWorkingTree() {
-    runJs("ide.soleOpenedProject().openTab()");
-
+  public void pullCurrentBranchAndEnsureCorrectRepositoryState() {
     runJs("ide.soleOpenedProject().checkoutBranch('allow-ownership-link')");
     runJs("ide.awaitNoBackgroundTask()");
+    // Remote tracking data is purposefully NOT set for this branch.
+    // Our plugin should infer the remote tracking branch based on its name.
     runJs("ide.soleOpenedProject().pullBranch('allow-ownership-link')");
     runJs("ide.awaitNoBackgroundTask()");
 
+    String localBranchHash = callJs("ide.soleOpenedProject().getHashOfCommitPointedByBranch('allow-ownership-link')");
+    String remoteBranchHash = callJs("ide.soleOpenedProject().getHashOfCommitPointedByBranch('origin/allow-ownership-link')");
+    Assert.assertEquals(remoteBranchHash, localBranchHash);
```

Comment:
*TODO: this is failing now*

Reply:
Note that our status shows `(behind origin)` for `allow-ownership-link`... so we can't pretend the pull isn't possible ;D

Reply:
```
com.virtuslab.gitmachete.uitest.UiTestSuite > pullCurrentBranchAndEnsureCorrectRepositoryState FAILED
    org.junit.ComparisonFailure: expected:<[87dd4e5b52ed363e2103fd33d72ca6bcf38d98d5]> but was:<[e754d1121917a3c2fd67b01947295ca8e6674240]>
```